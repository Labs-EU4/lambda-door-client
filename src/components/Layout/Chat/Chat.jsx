import React, { useState, createRef } from 'react';
import { sendMessage, closeChat } from '../../../state/actions/chat';
import { withFormik } from 'formik';
import { connect } from 'react-redux';
import { Avatar, Input, Form, Icon } from 'antd';
import {
  SendOutlined,
  ArrowsAltOutlined,
  ShrinkOutlined,
} from '@ant-design/icons';
import styled from 'styled-components';

const ChatCon = styled.div`
  display: inline-block;
  background: white;
  margin-right: 2em;
  max-width: 300px;
  width: 280px;
  border-top-right-radius: 10px;
  border-top-left-radius: 10px;
  -webkit-box-shadow: -1px 0px 13px -4px rgba(0, 21, 41, 0.51);
  -moz-box-shadow: -1px 0px 13px -4px rgba(0, 21, 41, 0.51);
  box-shadow: -1px 0px 13px -4px rgba(0, 21, 41, 0.51);

  &.isMinimized {
    #chat_body {
      display: none;
    }
    #chat_footer {
      display: none;
    }
  }
`;

const ChatHeader = styled.div`
  background: #bb1333;
  width: 100%;
  border-top-right-radius: 10px;
  border-top-left-radius: 10px;
  display: flex;
  justify-content: space-between;
  border-bottom: 1px solid rgba(217, 217, 217, 1);

  .top-chat {
    display: flex;
    align-items: center;
    margin-top: 0.5em;
    margin-bottom: 0.5em;
    margin-left: 1em;
    align-content: center;
    justify-content: space-between;

    h4 {
      margin: 0 0.6em;
      font-weight: 600;
      color: #fff;
    }
  }

  .chat_header_icon {
    display: flex;
    margin-right: 1em;
    align-items: center;
  }
`;

const ChatBody = styled.div`
  height: 250px;
  overflow-y: scroll;
  padding-bottom: 5em;
  display: flex;
  flex-direction: column;

  &::-webkit-scrollbar {
    width: 0em;
  }

  &::-moz-scrollbar-width {
    width: 0em;
  }

  &::-ms-scrollbar-width {
    width: 0em;
  }

  .chat_message {
    display: flex;
    background: #ebebeb;
    margin: 0.5em 0.6em;
    border-radius: 5px;
    font-size: 0.8rem;
    color: rgba(117, 117, 117, 1);
    width: fit-content;
    max-width: 70%;

    p {
      margin: 0.5em 0.7em;
    }

    &.move_right {
      align-self: flex-end;
    }
  }
`;

const ChatFooter = styled.div`
  background: #ffffff;
  position: fixed;
  bottom: 0;
  width: inherit;
  height: 42px;
  -webkit-box-shadow: 0px -7px 13px -13px rgba(0, 0, 0, 0.75);
  -moz-box-shadow: 0px -7px 13px -13px rgba(0, 0, 0, 0.75);
  box-shadow: 0px -7px 13px -13px rgba(0, 0, 0, 0.75);

  .ant-input-affix-wrapper {
    display: flex;
    justify-content: center;

    .ant-input {
      width: 95%;
      height: auto;
      margin: 0.3em;
    }
  }

  .ant-input-suffix {
    padding: 0 0.3em;
    color: #bb1333;
  }
`;

const Chat = ({
  chatState,
  values,
  touched,
  errors,
  handleChange,
  handleBlur,
  sendMessage,
  authState: {
    credentials: { id },
  },
  messages,
  chatID,
  closeChat,
  chat,
  form,
}) => {
  const [isMinimized, setIsMinimized] = useState(false);

  const minimizeChat = () => {
    setIsMinimized(!isMinimized);
  };

  const myRef = createRef();

  const handleSubmit = e => {
    e.preventDefault();
    form.validateFieldsAndScroll((err, values) => {
      if (!err) {
        sendMessage(values.message, chatID, id);
        form.resetFields();
        myRef.current.scrollIntoView({
          behavior: 'smooth',
          block: 'start',
        });
      } else {
        console.log('Enter Required Fields');
      }
    });
  };

  const { getFieldDecorator } = form;

  return (
    <>
      <ChatCon id="chat_con" className={isMinimized ? 'isMinimized' : ''}>
        <ChatHeader id="chat_header" style={{}}>
          <div className="top-chat">
            <Avatar style={{ color: '#f56a00', backgroundColor: '#fde3cf' }}>
              U
            </Avatar>
            <h4>
              {chat.fromUserID === id ? chat.toUserName : chat.fromUserName}
            </h4>
          </div>
          <div className="chat_header_icon">
            {isMinimized ? (
              <ArrowsAltOutlined
                onClick={minimizeChat}
                style={{ fontSize: '22px', color: '#fff' }}
              />
            ) : (
              <ShrinkOutlined
                onClick={minimizeChat}
                style={{ fontSize: '22px', color: '#fff' }}
              />
            )}
            <Icon
              type="close"
              style={{
                fontSize: '22px',
                color: '#fff',
                cursor: 'pointer',
                marginLeft: '10px',
              }}
              theme="outlined"
              onClick={e => closeChat(chatID)}
            />
          </div>
        </ChatHeader>

        <ChatBody id="chat_body">
          {messages.map((message, index) => {
            return (
              <div
                className={`chat_message ${
                  message.fromUserID === id ? 'move_right' : null
                }`}
                key={index}
              >
                <p>{message.message}</p>
                <div ref={myRef}></div>
              </div>
            );
          })}
        </ChatBody>

        <ChatFooter id="chat_footer">
          <Form onSubmit={handleSubmit}>
            <Form.Item>
              {getFieldDecorator('message', {
                rules: [
                  {
                    message: 'Enter valid message',
                  },
                  {
                    required: true,
                    message: 'Enter valid message',
                  },
                ],
              })(
                <Input
                  placeholder="Type a message..."
                  suffix={<SendOutlined onClick={handleSubmit} />}
                  name="message"
                  type="text"
                  autoComplete="off"
                />
              )}
            </Form.Item>
          </Form>
        </ChatFooter>
      </ChatCon>
    </>
  );
};

const ChatForm = Form.create({ name: 'text' })(Chat);

// const ChatForm = withFormik({
//   mapPropsToValues: () => ({ message: '' }),

//   // Custom sync validation
//   validate: values => {
//     const errors = {};

//     if (!values.message) {
//       errors.message = 'Required';
//     }

//     return errors;
//   },

//   handleSubmit: (values, { setSubmitting, setFieldValue, props }) => {
//     props.sendMessage(
//       values.message,
//       props.chatID,
//       props.authState.credentials.id
//     );
//     setFieldValue('message', '');
//   },

//   displayName: 'BasicForm',
// })(Chat);

export default connect(state => state, { sendMessage, closeChat })(ChatForm);
