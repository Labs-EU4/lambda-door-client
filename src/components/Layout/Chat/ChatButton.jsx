import React from 'react';
// import { getMessages } from '../../../utils/firebase';

import { getMessages } from '../../../state/actions/chat';

import { connect } from 'react-redux';

const ChatButton = ({
  chatState,
  getMessages,
  authState: {
    credentials: { id },
  },
  toUserID,
}) => {
  const chatClick = e => {
    // getMessages(id, toUserID);
    // create new chat
  };

  return <div onClick={e => chatClick()}>Chat with me</div>;
};

export default connect(state => state, { getMessages })(ChatButton);
