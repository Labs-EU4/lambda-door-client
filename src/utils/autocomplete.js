/* eslint-disable react/jsx-no-undef */
import React, { useState } from 'react';
import { Form, Input, AutoComplete, Button } from 'antd';
import AddCompanyModal from '../components/AddReview/AddCompanyModal';

const { Option } = AutoComplete;

const renderOption = Item => {
  return (
    // eslint-disable-next-line react/jsx-filename-extension
    <Option key={Item.name} text={Item.name}>
      <p>{Item.name}</p>
    </Option>
  );
};

const AutoCompleted = ({ label, dataSource, placeholder, onChange }) => {
  const [addingCompany, setAddingCompany] = useState(false);
  const [value, setValue] = useState('');

  const options = dataSource.map(renderOption).concat([
    <Option key="all" value="" className="show-all">
      <Button type="button" onClick={() => setAddingCompany(true)}>
        Add Company
      </Button>
    </Option>,
  ]);

  return (
    <>
      <AddCompanyModal
        visible={addingCompany}
        setAddingCompany={setAddingCompany}
        value={value}
      />
      <Form.Item label={label}>
        <AutoComplete
          className="certain-category-search"
          dropdownClassName="certain-category-search-dropdown"
          dropdownMatchSelectWidth={false}
          dropdownStyle={{ width: 300 }}
          size="default"
          style={{ width: '100%' }}
          dataSource={options}
          optionLabelProp="value"
          filterOption={(inputValue, option) => {
            if (
              option.key.toLowerCase().includes(inputValue.toLowerCase()) ||
              option.key.toLowerCase() === 'all'
            ) {
              return true;
            }
            return false;
          }}
          onChange={e => {
            if (e) {
              onChange(e);
              setValue(e);
            }
          }}
        >
          <Input size="default" placeholder={placeholder} />
        </AutoComplete>
      </Form.Item>
    </>
  );
};

export default AutoCompleted;
