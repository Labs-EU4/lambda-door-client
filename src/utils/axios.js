import axios from 'axios';

axios.defaults.baseURL = `${process.env.REACT_APP_BACKEND_URL}`

export default function withAuth() {
  const token = localStorage.getItem('token');
  const instance = axios.create({
    headers: {
      'Content-Type': 'application/json',
      Authorization: `${token}`,
    },
  });

  return instance;
}
