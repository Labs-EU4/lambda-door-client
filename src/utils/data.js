export const companies = [
  {
    id: 1,
    name: 'Accenture',
    website: 'www.accenture.com.',
    location: 'Atlanta, GA',
    type: 'Business',
    logo: '',
    description: '',
  },
  {
    id: 2,
    name: 'Anthem, Inc.',
    website: 'https://www.antheminc.com',
    location: 'Atlanta, GA',
    type: 'Health',
    logo: '',
    description: '',
  },
  {
    id: 3,
    name: 'Ahalogy',
    website: 'https://www.ahalogy.com',
    location: 'Cincinnati, OH',
    type: 'Business',
    logo: '',
    description: '',
  },
  {
    id: 4,
    name: 'Bad Rabbit',
    website: 'https://www.badrabbit.com',
    location: 'Portland, OR',
    type: 'Technology',
    logo: '',
    description: '',
  },
  {
    id: 5,
    name: 'DoNotPay Inc',
    website: 'https://donotpay.com',
    location: 'Technology',
    type: 'rowValue3',
    logo: '',
    description: '',
  },
  {
    id: 6,
    name: 'Newfront Insurance',
    website: 'https://www.newfrontinsurance.com',
    location: 'San Francisco, CA',
    type: 'insurance',
    logo: '',
    description: '',
  },
  {
    id: 7,
    name: 'Unspun',
    website: 'https://www.unspuntech.com/',
    location: 'San Francisco, CA',
    type: 'Commerce',
    logo: '',
    description: '',
  },
  {
    id: 8,
    name: 'rideOS',
    website: 'https://rideos.ai',
    location: 'San Francisco, CA',
    type: 'Technology',
    logo: '',
    description: '',
  },
  {
    id: 9,
    name: 'Mesa',
    website: '',
    location: 'Austin, TX',
    type: 'Software',
    logo: '',
    description: '',
  },
  {
    id: 10,
    name: 'Paystack',
    website: 'https://paystack.com/',
    location: 'Lagos, NG',
    type: 'FinTech',
    logo: '',
    description: '',
  },
  {
    id: 11,
    name: 'BuyCoins',
    website: 'https://buycoins.africa/',
    location: 'Lagos, NG',
    type: 'Crypto',
    logo: '',
    description: '',
  },
  {
    id: 12,
    name: 'Chipper Cash',
    website: 'https://chippercash.com/',
    location: 'Lagos, NG',
    type: 'FinTech',
    logo: '',
    description: '',
  },
  {
    id: 13,
    name: 'Cowrywise',
    website: 'https://www.cowrywise.com/',
    location: 'Lagos, NG',
    type: 'FinTech',
    logo: '',
    description: '',
  },
  {
    id: 14,
    name: 'Others',
    website: '',
    location: '',
    type: '',
    logo: '',
    description: '',
  },
];
export const salaryReviews = [
  {
    id: 1,
    user_id: 1,
    company_id: 1,
    description: 'Accenture Programmer',
    salary: 95000,
    currency: 'USD',
    is_accepting_questions: 0,
  },
  {
    id: 2,
    user_id: 2,
    company_id: 3,
    description: 'Software Engineer',
    salary: 98000,
    currency: 'USD',
    is_accepting_questions: 0,
  },
  {
    id: 3,
    user_id: 3,
    company_id: 10,
    description: 'Junior Developer',
    salary: 3000000,
    currency: 'NGN',
    is_accepting_questions: 1,
  },
  {
    id: 4,
    user_id: 4,
    company_id: 1,
    description: 'Accenture Technical Specialist',
    salary: 85000,
    currency: 'USD',
    is_accepting_questions: 1,
  },
  {
    id: 5,
    user_id: 5,
    company_id: 13,
    description: 'Software Engineer',
    salary: '4800000',
    currency: 'NGN',
    is_accepting_questions: 0,
  },
];
export const interviewReviews = [
  {
    id: 1,
    user_id: 1,
    company_id: 1,
    company_name: 'Microsoft',
    text:
      'Six rounds of phone/tech interviews over a long time period. It seemed a bit scattered and could have been way more efficient. I felt like some of the interviews got repetitive. The entire process took three months and I was able to interview at another company and find a job whilst waiting to hear back.',
  },
  {
    id: 2,
    user_id: 2,
    company_id: 5,
    company_name: 'Google',
    text:
      'Very pleasant recruiters. First a phone interview that took about 15 minutes. Followed by two Skype interviews; first with a recruiter after that with a manager.Basic trouble shooting questions. They want to really get to know you as a person. Just be yourself and do not be afraid to be geeky/nerdy! Make sure you have a desk and dedicated home phone line.',
  },
  {
    id: 3,
    user_id: 4,
    company_id: 11,
    company_name: 'Amazon',
    text:
      'Completed their developer challenge and received an email offering an interview. A few preliminary questions via email. Scheduled a phone interview. The recruiter did not call me, and did not respond to my email asking why they missed the scheduled interview. I am not sure what happened, but after scheduling the interview, there is no reason to not show up.',
  },
  {
    id: 4,
    user_id: 3,
    company_id: 6,
    company_name: 'Twitter',
    text:
      'There is one phone interview for an hour. If you clear this interview, you will be invited for an onsite interview. There are five interviews onsite, all in one day. Interviews are in a casual environment. After the first two interviews, you take a break and are escorted for lunch.',
  },
  {
    id: 5,
    user_id: 5,
    company_id: 12,
    company_name: 'Slack',
    text:
      "They initially called me in for a technical interview and I didn't hear anything back for three weeks. After this I was surprised to get another call saying that I had been invited for a behavioural interview. In total the process took over two months. The interviews were both with a panel and were very thorough but the atmosphere was relaxed and friendly.",
  },

  {
    company_id: 21,
    company_name: 'Novosco',
    text:
      'I was invited to a very informal chat in the first place and shown around the office. Two weeks later I got a technical interview and I met some more people from the office. The interview lasted for an hour and had both easy and difficult questions. Two days later I got a phone call saying that I had got the job.',
  },
];

export const allInterests = [
  'Software Engineer',
  'Front End',
  'Back End',
  'Full Stack',
  'Data Science',
  'Machine Learning',
  'User Experience',
  'Mobile Development',
  'Product Manager',
];
export const jobCategories = [
  {
    id: 1,
    name: 'Software Engineer',
  },
  {
    id: 2,
    name: 'Front End',
  },
  {
    id: 3,
    name: 'Back End',
  },
  {
    id: 4,
    name: 'Full Stack',
  },
  {
    id: 5,
    name: 'Data Science',
  },
  {
    id: 6,
    name: 'Machine Learning',
  },
  {
    id: 7,
    name: 'User Experience',
  },
];
export const currencies = [
  {
    id: 1,
    name: 'USD',
    symbol: '$',
    full_name: 'USD Dollars',
  },
  {
    id: 2,
    name: 'EUR',
    symbol: '',
    full_name: 'EUROS',
  },
  {
    id: 3,
    name: 'GBP',
    symbol: '',
    full_name: 'British Pounds',
  },
  {
    id: 4,
    name: 'NAIRA',
    symbol: '',
    full_name: 'Nigerian Naira',
  },
];
