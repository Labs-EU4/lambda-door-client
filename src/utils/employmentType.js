export const employmentType = [
  {
    id: 1,
    name: 'Full-time',
  },
  {
    id: 2,
    name: 'Contract',
  },
  {
    id: 3,
    name: 'Part-time',
  },
  {
    id: 4,
    name: 'Internship',
  },
];
