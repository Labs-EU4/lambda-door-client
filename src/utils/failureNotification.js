import { notification } from 'antd';

const failureNotification = message => {
  notification.open({
    message: 'Something went wrong',
    description: message,
  });
};

export default failureNotification;
