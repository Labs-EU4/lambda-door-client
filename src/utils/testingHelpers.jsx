import React from 'react';
import { createStore, applyMiddleware, compose } from 'redux';
import { Provider } from 'react-redux';
import { Router, Route } from 'react-router-dom';
import { render } from '@testing-library/react';
import thunk from 'redux-thunk';
import { createMemoryHistory } from 'history';
import rootReducer from '../state/reducers/index';

export const TestingRouter = ({
  ComponentWithRedirection,
  RedirectUrl,
  MainRoute,
}) => (
  <>
    <Route
      path={MainRoute}
      exact={true}
      render={props => {
        return <ComponentWithRedirection {...props} />;
      }}
    />
    <Route path={RedirectUrl} render={() => <div>{RedirectUrl}</div>} />
  </>
);

export const renderWithRedux = (
  component,
  {
    initialState,
    store = createStore(
      rootReducer,
      initialState,
      compose(applyMiddleware(thunk))
    ),
  } = {},
  {
    route = '/',
    history = createMemoryHistory({ initialEntries: [route] }),
  } = {}
) => {
  return {
    ...render(
      <Provider store={store}>
        <Router history={history}>{component}</Router>
      </Provider>
    ),
    store,
    history,
  };
};
