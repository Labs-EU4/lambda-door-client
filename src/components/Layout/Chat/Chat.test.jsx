import React from 'react';
import * as rtl from '@testing-library/react';
import Chat from './Chat';
import '@testing-library/jest-dom/extend-expect';
import { renderWithRedux } from '../../../utils/testingHelpers';

beforeEach(rtl.cleanup);

const chats = [
  {
    toUserID: 1,
    fromUserID: 6,
    fromUserName: 'Rodrigo Graça',
    toUserName: 'Lisa Wilton',
    messages: [
      {
        fromUserID: 6,
        message: 'some new message 1',
      },
      {
        fromUserID: 6,
        message: 'some new message 2',
      },
    ],
    open: true,
  },
  {
    toUserID: 1,
    fromUserID: 6,
    fromUserName: 'Rodrigo Graça',
    toUserName: 'Lisa Wilton',
    messages: [
      {
        fromUserID: 6,
        message: 'what new message?',
      },
      {
        fromUserID: 6,
        message: 'wowza',
      },
    ],
    open: true,
  },
];

const messages = chats.map(messages => messages);

const chatMessages = [
  {
    fromUserID: 6,
    message: 'wowza',
  },
];

const sendMessage = jest.fn();
const closeChat = jest.fn();
const markAsRead = jest.fn();

describe('Chat Interface', () => {
  test('should render the Chat Interface without crashing', () => {
    const chatElement = renderWithRedux(
      <Chat
        sendMessage={sendMessage}
        closeChat={closeChat}
        markAsRead={markAsRead}
        messages={messages}
        chat={chats}
      />
    );
    expect(chatElement.baseElement).toMatchSnapshot();
  });

  test('should display the correct messages', () => {
    const chatElement = renderWithRedux(
      <Chat
        sendMessage={sendMessage}
        closeChat={closeChat}
        markAsRead={markAsRead}
        messages={chatMessages}
        chat={chats}
      />
    ).queryAllByText(/wowza/i);
    console.log(`chatElement`, chatElement);

    // expect(chatElement[0].textContent).toBeInTheDocument();

    // const message = .queryAllByText(/message 1.1/i);
    // // expect(message[0].textContent).toBe('');
    // expect(message[0].textContent).toBeInTheDocument();
  });
});
