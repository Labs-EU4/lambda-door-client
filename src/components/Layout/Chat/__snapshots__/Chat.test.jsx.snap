// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chat Interface should render the Chat Interface without crashing 1`] = `
<body>
  <div>
    <div
      class="sc-bdVaJa kdOWNY"
      id="chat_con"
    >
      <div
        class="sc-bwzfXH Koijq"
        id="chat_header"
      >
        <div
          class="top-chat"
        >
          <span
            class="ant-avatar ant-avatar-circle"
            style="color: rgb(245, 106, 0); background-color: rgb(253, 227, 207);"
          >
            <span
              class="ant-avatar-string"
              style="transform: scale(1) translateX(-50%);"
            >
              U
            </span>
          </span>
          <h4 />
        </div>
        <div
          class="chat_header_icon"
        >
          <span
            aria-label="shrink"
            class="anticon anticon-shrink"
            role="img"
            style="font-size: 22px; color: rgb(255, 255, 255);"
            tabindex="-1"
          >
            <svg
              aria-hidden="true"
              class=""
              data-icon="shrink"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M881.7 187.4l-45.1-45.1a8.03 8.03 0 00-11.3 0L667.8 299.9l-54.7-54.7a7.94 7.94 0 00-13.5 4.7L576.1 439c-.6 5.2 3.7 9.5 8.9 8.9l189.2-23.5c6.6-.8 9.3-8.8 4.7-13.5l-54.7-54.7 157.6-157.6c3-3 3-8.1-.1-11.2zM439 576.1l-189.2 23.5c-6.6.8-9.3 8.9-4.7 13.5l54.7 54.7-157.5 157.5a8.03 8.03 0 000 11.3l45.1 45.1c3.1 3.1 8.2 3.1 11.3 0l157.6-157.6 54.7 54.7a7.94 7.94 0 0013.5-4.7L447.9 585a7.9 7.9 0 00-8.9-8.9z"
              />
            </svg>
          </span>
          <i
            aria-label="icon: close"
            class="anticon anticon-close"
            style="font-size: 22px; color: rgb(255, 255, 255); cursor: pointer; margin-left: 10px;"
            tabindex="-1"
          >
            <svg
              aria-hidden="true"
              class=""
              data-icon="close"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M563.8 512l262.5-312.9c4.4-5.2.7-13.1-6.1-13.1h-79.8c-4.7 0-9.2 2.1-12.3 5.7L511.6 449.8 295.1 191.7c-3-3.6-7.5-5.7-12.3-5.7H203c-6.8 0-10.5 7.9-6.1 13.1L459.4 512 196.9 824.9A7.95 7.95 0 0 0 203 838h79.8c4.7 0 9.2-2.1 12.3-5.7l216.5-258.1 216.5 258.1c3 3.6 7.5 5.7 12.3 5.7h79.8c6.8 0 10.5-7.9 6.1-13.1L563.8 512z"
              />
            </svg>
          </i>
        </div>
      </div>
      <div
        class="sc-htpNat gvFqbI"
        id="chat_body"
      >
        <div
          class="chat_message null"
        >
          <p />
          <div />
        </div>
        <div
          class="chat_message null"
        >
          <p />
          <div />
        </div>
      </div>
      <div
        class="sc-bxivhb hNeMQD"
        id="chat_footer"
      >
        <form
          class="ant-form ant-form-horizontal"
        >
          <div
            class="ant-row ant-form-item"
          >
            <div
              class="ant-col ant-form-item-control-wrapper"
            >
              <div
                class="ant-form-item-control"
              >
                <span
                  class="ant-form-item-children"
                >
                  <span
                    class="ant-input-affix-wrapper"
                  >
                    <input
                      autocomplete="off"
                      class="ant-input"
                      data-__field="[object Object]"
                      data-__meta="[object Object]"
                      id="text_message"
                      name="message"
                      placeholder="Type a message..."
                      type="text"
                      value=""
                    />
                    <span
                      class="ant-input-suffix"
                    >
                      <span
                        aria-label="send"
                        class="anticon anticon-send"
                        role="img"
                        tabindex="-1"
                      >
                        <svg
                          aria-hidden="true"
                          class=""
                          data-icon="send"
                          fill="currentColor"
                          focusable="false"
                          height="1em"
                          viewBox="64 64 896 896"
                          width="1em"
                        >
                          <defs>
                            <style />
                          </defs>
                          <path
                            d="M931.4 498.9L94.9 79.5c-3.4-1.7-7.3-2.1-11-1.2a15.99 15.99 0 00-11.7 19.3l86.2 352.2c1.3 5.3 5.2 9.6 10.4 11.3l147.7 50.7-147.6 50.7c-5.2 1.8-9.1 6-10.3 11.3L72.2 926.5c-.9 3.7-.5 7.6 1.2 10.9 3.9 7.9 13.5 11.1 21.5 7.2l836.5-417c3.1-1.5 5.6-4.1 7.2-7.1 3.9-8 .7-17.6-7.2-21.6zM170.8 826.3l50.3-205.6 295.2-101.3c2.3-.8 4.2-2.6 5-5 1.4-4.2-.8-8.7-5-10.2L221.1 403 171 198.2l628 314.9-628.2 313.2z"
                          />
                        </svg>
                      </span>
                    </span>
                  </span>
                </span>
              </div>
            </div>
          </div>
        </form>
      </div>
    </div>
  </div>
</body>
`;
